device/cuda/include/traccc/cuda/utils/barrier.hpp

```diff
@@ -0,0 +1,23 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/definitions/qualifiers.hpp"
+
+namespace traccc::cuda {
+
+struct barrier {
+    TRACCC_DEVICE
+    void blockBarrier() { __syncthreads(); }
+
+    TRACCC_DEVICE
+    bool blockOr(bool predicate) { return __syncthreads_or(predicate); }
```

Comment:
Can we get these functions to support thread masks?

Reply:
I'm not sure that would really be the way to go here. I understand it's usefulness, but the whole point of this class was to have the same interface as it's SYCL twin, and SYCL does not provide such a functionality AFAIK

Reply:
Never mind I need another coffee, block synchronization does not take masks, but this problem will come back later when we add workgroup/warp synchronization.